logzio-log-sender/src/main/java/io/logz/sender/LogzioBaseJsonFormatter.java

```diff
@@ -0,0 +1,79 @@
+package io.logz.sender;
+
+import com.google.common.base.Splitter;
+import com.google.gson.JsonObject;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created by MarinaRazumovsky on 18/12/2016.
+ */
+public abstract class LogzioBaseJsonFormatter<T> {
+
+    public static final String TIMESTAMP = "@timestamp";
+    public static final String LOGLEVEL = "loglevel";
+    public static final String MARKER = "marker";
+    public static final String MESSAGE = "message";
+    public static final String LOGGER = "logger";
+    public static final String THREAD = "thread";
+    public static final String EXCEPTION = "exception";
+    public static final String SOURCE = "source";
+
+    public static final String SOURCE_FILE ="file";
+    public static final String SOURCE_LINE_NUMBER = "lineNumber";
+    public static final String SOURCE_CLASS ="class";
+    public static final String SOURCE_METHOD = "method";
+
+
+    private final Map<String, String> additionalFieldsMap;
+
+    private final Set<String> reservedFields;
+
+    public LogzioBaseJsonFormatter(String additionalFields, boolean addHostname, ILogzioStatusReporter reporter, Set<String> fieldsSet){
```

Comment:
I think this class can be omitted. The Sender can expose a method called send(...) that contains all the parameters you need to format a JSON Log Message: ```long timestamp, String logLevelName, String message, String loggerName, String threadName, Optional<Map<String, String>> mdcPropertyMap, String exceptionAsString)```
Thus the logic of extracting those fields from the framework logging event object will remain in the cotext of the specific appenders.
This will save us from: Abstract class which are generally good thing to avoid, Generics - again, avoid if you can.